src/components/RenameHeader.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { FC, ReactNode } from "react";
+import { Button, Input, Tooltip } from "@canonical/react-components";
+import SubmitButton from "components/SubmitButton";
+import { FormikProps } from "formik/dist/types";
+
+export interface RenameHeaderValues {
+  name: string;
+  isRenaming: boolean;
+}
+
+interface Props {
+  name: string;
+  parentItem: ReactNode;
+  centerControls?: ReactNode;
+  controls?: ReactNode;
+  isLoaded: boolean;
+  renameDisabledReason?: string;
+  formik: FormikProps<RenameHeaderValues>;
+}
+
+const RenameHeader: FC<Props> = ({
+  name,
+  parentItem,
+  centerControls,
+  controls,
+  isLoaded,
+  formik,
+  renameDisabledReason,
+}) => {
+  const canRename = renameDisabledReason === undefined;
+
+  return (
+    <div className="p-panel__header rename-header">
+      <h1 className="u-off-screen">{name}</h1>
+      {isLoaded ? (
+        <div className="p-panel__title">
+          <nav
+            key="breadcrumbs"
+            className="p-breadcrumbs"
+            aria-label="Breadcrumbs"
+          >
+            <ol className="p-breadcrumbs__items">
+              <li className="p-breadcrumbs__item">{parentItem}</li>
+              {formik.values.isRenaming ? (
+                <li className="p-breadcrumbs__item rename">
+                  <Input
+                    autoFocus
+                    id="name"
+                    name="name"
+                    className="name-input"
+                    onBlur={formik.handleBlur}
+                    onChange={formik.handleChange}
+                    value={formik.values.name}
+                    error={formik.touched.name ? formik.errors.name : null}
+                    onKeyUp={(e) =>
+                      e.key === "Enter" && void formik.submitForm()
+                    }
+                    type="text"
+                  />
+                  <div>
+                    <Button
+                      appearance="base"
+                      className="cancel"
+                      onClick={() => formik.setFieldValue("isRenaming", false)}
```

Comment:
Can we also reset the value in the input field when we cancel a rename?

![Peek 2023-04-20 17-58](https://user-images.githubusercontent.com/56583786/233426185-10b528d9-8a91-4489-9327-5b0a2d0f00f7.gif)